query-languages/m/m-spec-operators.md

```diff
@@ -1263,7 +1263,7 @@ nullable-primitive-type:_<br/>
 
 Type compatibility, as supported by the `is` operator, is a subset of [general type compatibility](m-spec-types.md) and is defined using the following rules:
 
-* If `x` is null then it is compatible iff y is a nullable type or the type `any`.
+* If `x` is null then it is compatible if `y` is a nullable type, the type `null` or the type `any`.
```

Comment:
Isn't this redundant? The type `null` appears to be nullable.

Type.IsNullable(type null) => true

Reply:
Also, I'm not sure if the "iff" is a typo (although it's probably confusing, if it isn't). It means "if and only if". I'm not sure it's necessary here.

Do we have any other uses of "iff" in the language spec?

Reply:
Thanks, @ehrenMSFT.

> Isn't this redundant? The type `null` appears to be nullable.
>
> Type.IsNullable(type null) => true

Wouldn't `null` be in the same category as `any` (which was listed here before this edit was proposed)? 

The standard library function `Type.IsNullable` returns true for both. However, based on a strict reading of the language specification, maybe neither technically are what the spec classifies as "nullable types"?

At the top of this page, in the listing of type groups , types `any` and `null` are included in the first grouping, which covers primitive types. In contrast, nullable types are mentioned separately towards the bottom of the groupings list. To me, strictly speaking, this suggests that they are distinct.

Another way to look at it: Say, for some strange reason, we decided to get rid of nullable types altogether from M, so we removed the bullet for nullable types. Even with this hypothetical removal, types `any` and `null` would still remain in the language, because their existence is not dependent on nullable types being present.

On the other hand, if we are more generous (less strict) in our reading of the spec, and say that `null` is a nullable type (and so this edit is redundant), then it seems like the same determination should apply to `any` (so the mention of it should be edited out of the line in question).

What do you think?

Reply:
> Also, I'm not sure if the "iff" is a typo (although it's probably confusing, if it isn't). It means "if and only if". I'm not sure it's necessary here.
> 
> Do we have any other uses of "iff" in the language spec?

Did a quick search of the other spec pages and didn't find other uses of it.

Reply:
Ok. I guess it's not bad to be explicit about the basic built-in types here. How about:

```suggestion
* If `x` is null then it is compatible if `y` is the type `any`, the type `null`, or a nullable type.
```

Reply:
Thanks!